f5lbaasdriver/v2/bigip/service_builder.py

```diff
@@ -277,20 +281,26 @@ def _populate_member_network(self, context, member, network):
         member['vxlan_vteps'] = []
         member['gre_vteps'] = []
 
-        if 'provider:network_type' in network:
-            net_type = network['provider:network_type']
-            if net_type == 'vxlan':
-                if 'binding:host_id' in member['port']:
-                    host = member['port']['binding:host_id']
-                    member['vxlan_vteps'] = self._get_endpoints(
-                        context, 'vxlan', host)
-            if net_type == 'gre':
-                if 'binding:host_id' in member['port']:
-                    host = member['port']['binding:host_id']
-                    member['gre_vteps'] = self._get_endpoints(
-                        context, 'gre', host)
-        if 'provider:network_type' not in network:
-            network['provider:network_type'] = 'undefined'
+        agent_config = {}
+        segment_data = self.disconnected_service.get_network_segment(
+            context, agent_config, network)
+        if segment_data:
+            network['provider:segmentation_id'] = \
+                segment_data.get('segmentation_id', None)
+            network['provider:network_type'] = \
+                segment_data.get('network_type', None)
+
+        net_type = network.get('provider:network_type', "undefined")
+        if net_type == 'vxlan':
+            if 'binding:host_id' in member['port']:
+                host = member['port']['binding:host_id']
+                member['vxlan_vteps'] = self._get_endpoints(
+                    context, 'vxlan', host)
+        if net_type == 'gre':
+            if 'binding:host_id' in member['port']:
+                host = member['port']['binding:host_id']
+                member['gre_vteps'] = self._get_endpoints(
+                    context, 'gre', host)
         if 'provider:segmentation_id' not in network:
             network['provider:segmentation_id'] = 0
```

Comment:
Is it safer to set network['provider:segmentation_id'] = None (instead of 0) so the agent can punt on an incomplete network?   The new design will return pending = True for a missing seg_id, right?